packages/nc-gui/components/smartsheet/calendar/DayView/DateField.vue

```diff
@@ -0,0 +1,238 @@
+          position = 'none'
+        } else if (isSelectedDay(startDate) && isAfterSelectedDay(endDate)) {
+          position = 'leftRounded'
+        } else if (isBeforeSelectedDay(startDate) && isSelectedDay(endDate)) {
+          position = 'rightRounded'
+        } else {
+          position = 'none'
+        }
+
+        recordsByRange.push({
+          ...record,
+          rowMeta: {
+            ...record.rowMeta,
+            position,
+            style,
+            range: range as any,
+          },
+        })
+      }
+    } else if (fromCol) {
+      for (const record of formattedData.value) {
+        dayRecordCount++
+        recordsByRange.push({
+          ...record,
+          rowMeta: {
+            ...record.rowMeta,
+            range: range as any,
+            style: {
+              width: '100%',
+              left: '0',
+              top: `${(dayRecordCount - 1) * perRecordHeight}px`,
+            },
+            position: 'rounded',
+          },
+        })
+      }
+    }
+  })
+  return recordsByRange
+})
+
+const dragElement = ref<HTMLElement | null>(null)
+
+const hoverRecord = ref<string | null>(null)
+
+const dropEvent = (event: DragEvent) => {
+  if (!isUIAllowed('dataEdit')) return
+  event.preventDefault()
+  const data = event.dataTransfer?.getData('text/plain')
+  if (data) {
+    const {
+      record,
+    }: {
+      record: Row
+      initialClickOffsetY: number
+      initialClickOffsetX: number
+    } = JSON.parse(data)
+
+    const fromCol = record.rowMeta.range?.fk_from_col
+    const toCol = record.rowMeta.range?.fk_to_col
+
+    if (!fromCol) return
+
+    const newStartDate = dayjs(selectedDate.value).startOf('day')
+
+    let endDate
+
+    const newRow = {
+      ...record,
+      row: {
+        ...record.row,
+        [fromCol.title!]: dayjs(newStartDate).format('YYYY-MM-DD HH:mm:ssZ'),
+      },
+    }
+
+    const updateProperty = [fromCol.title!]
+
+    if (toCol) {
+      const fromDate = record.row[fromCol.title!] ? dayjs(record.row[fromCol.title!]) : null
+      const toDate = record.row[toCol.title!] ? dayjs(record.row[toCol.title!]) : null
+
+      if (fromDate && toDate) {
+        endDate = dayjs(newStartDate).add(toDate.diff(fromDate, 'day'), 'day')
+      } else if (fromDate && !toDate) {
+        endDate = dayjs(newStartDate).endOf('day')
+      } else if (!fromDate && toDate) {
+        endDate = dayjs(newStartDate).endOf('day')
+      } else {
+        endDate = newStartDate.clone()
+      }
+      newRow.row[toCol.title!] = dayjs(endDate).format('YYYY-MM-DD HH:mm:ssZ')
+      updateProperty.push(toCol.title!)
+    }
+
+    if (!newRow) return
+
+    if (dragElement.value) {
+      formattedData.value = formattedData.value.map((r) => {
+        const pk = extractPkFromRow(r.row, meta.value!.columns!)
+
+        if (pk === extractPkFromRow(newRow.row, meta.value!.columns!)) {
+          return newRow
+        }
```

Comment:
1. `extractPkFromRow(newRow.row, meta.value!.columns!)` doesn't change in the loop. please put it outside the loop to avoid unnecessary computation.
2. rename `r`. it took me a several seconds to figure out what it is
3. L148 new line is not necessary.

apply to other similar code